hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -317,7 +315,13 @@ private void ensurePartitionLoadedCorrectly(String partition) {
    * @throws IOException
    */
   protected FileStatus[] listPartition(Path partitionPath) throws IOException {
-    return metaClient.getFs().listStatus(partitionPath);
+    // Create the path if it does not exist already
+    if (!metaClient.getFs().exists(partitionPath)) {
+      metaClient.getFs().mkdirs(partitionPath);
```

Comment:
@nsivabalan hi, may i ask why we need to create dir when not exists. This behavior have a problem after drop partition, clean action will delete the dir. But this dropped partition maybe listed again, and will result in a empty dir, can we just `return new FileStatus[0]; ` here without create new dir.